otp.js

```diff
@@ -19,6 +21,16 @@ const api = new Messente.OmnimessageApi();
 
 const getOTP = () => crypto.randomInt(0,1000000).toString().padStart(6,'0')
 
+const cacheCountryRequest = async (ipAddr, country) => {
+    const count = await redis.hIncrBy('country_requests', country, 1)
+    await redis.rPush(country, ipAddr)
```

Comment:
I don't think we need to store this data -- we just need to store enough data to rate limit

Reply:
What do you think about following 

```
    const minuteKey = `minute:${countryCode}:${ipAddresses}`;
    const hourKey = `hour:${countryCode}:${ipAddresses}`;

    const countMinut = await redis.hIncrBy('country_requests_minutes', minuteKey, 1);
    const countHour = await redis.hIncrBy('country_requests_hours', hourKey, 1);

    await redis.expire('country_request_minutes', 60)
    await redis.expire('country_requests_hour', 3600) 
    if(countMinut > MAX_COUNTRY_ATTEMP_PER_MINUTE || countHour > MAX_COUNTRY_ATTEMP_PER_HOUR) {
        throw new Error('Too many recent attempts from country ', country) 
    }
```

Reply:
Looks great!

Reply:
Actually, minuteKey and hourKey seem based on a country/IP pair when they should be based on just country. Let's remove IP for now and we can add it later if needed. I think as you mentioned, ipv6 adds complication as well (it is also trivial to make many IPs so we'd need to block IPv6). So I think let's just change these two lines to:
 ``` 
 const minuteKey = `minute:${countryCode}`;
  const hourKey = `hour:${countryCode}`;
  ```
and then we're good to go.